pkg/conversion/capi2mapi/aws_test.go

```diff
@@ -18,75 +18,240 @@ package capi2mapi
+			expectedWarnings: []string{},
+		}),
+
+		Entry("With unsupported httpTokens", awsCAPI2MAPIMachineConversionInput{
+			awsClusterBuilder: awsCAPIAWSClusterBase,
+			awsMachineBuilder: awsCAPIAWSMachineBase.
+				WithInstanceMetadataOptions(&capav1.InstanceMetadataOptions{
+					HTTPTokens: "unsupported",
+				}),
+			machineBuilder:   awsCAPIMachineBase,
+			expectedErrors:   []string{"spec.instanceMetadataOptions.httpTokens: Invalid value: \"unsupported\": unable to convert httpTokens state, unknown value"},
+			expectedWarnings: []string{},
+		}),
+
+		Entry("With unsupported instanceMetadataTags", awsCAPI2MAPIMachineConversionInput{
+			awsClusterBuilder: awsCAPIAWSClusterBase,
+			awsMachineBuilder: awsCAPIAWSMachineBase.
+				WithInstanceMetadataOptions(&capav1.InstanceMetadataOptions{
+					InstanceMetadataTags: capav1.InstanceMetadataEndpointStateEnabled,
+				}),
+			machineBuilder:   awsCAPIMachineBase,
+			expectedErrors:   []string{"spec.instanceMetadataOptions.instanceMetadataTags: Invalid value: \"enabled\": instanceMetadataTags values other than \"disabled\" are not supported"},
+			expectedWarnings: []string{},
+		}),
+
+		// Test case for multiple metadata-related fields
+		Entry("With multiple unsupported metadata options", awsCAPI2MAPIMachineConversionInput{
+			awsClusterBuilder: awsCAPIAWSClusterBase,
+			awsMachineBuilder: awsCAPIAWSMachineBase.
+				WithInstanceMetadataOptions(&capav1.InstanceMetadataOptions{
+					HTTPEndpoint:            capav1.InstanceMetadataEndpointStateDisabled,
+					HTTPPutResponseHopLimit: 2,
+					HTTPTokens:              "unsupported",
+					InstanceMetadataTags:    capav1.InstanceMetadataEndpointStateEnabled,
+				}),
+			machineBuilder: awsCAPIMachineBase,
+			expectedErrors: []string{
+				"spec.instanceMetadataOptions.httpTokens: Invalid value: \"unsupported\": unable to convert httpTokens state, unknown value, " +
+					"spec.instanceMetadataOptions.httpEndpoint: Invalid value: \"disabled\": httpEndpoint values other than \"enabled\" are not supported, " +
+					"spec.instanceMetadataOptions.httpPutResponseHopLimit: Invalid value: 2: httpPutResponseHopLimit values other than 1 are not supported, " +
+					"spec.instanceMetadataOptions.instanceMetadataTags: Invalid value: \"enabled\": instanceMetadataTags values other than \"disabled\" are not supported"},
+			expectedWarnings: []string{},
+		}),
+	)
+
+	var _ = DescribeTable("capi2mapi AWS convert CAPI MachineSet/InfraMachineTemplate/InfraCluster to MAPI MachineSet",
+		func(in awsCAPI2MAPIMachinesetConversionInput) {
+			_, warns, err := FromMachineSetAndAWSMachineTemplateAndAWSCluster(
+				in.machineSetBuilder.Build(),
+				in.awsMachineTemplateBuilder.Build(),
+				in.awsClusterBuilder.Build(),
+			).ToMachineSet()
+			Expect(err).To(matchers.ConsistOfMatchErrorSubstrings(in.expectedErrors),
```

Comment:
another nit: we could helper-func this expect block, as we use it twice

Reply:
The failure comment is different though, not sure it is worth it :)

Reply:
I'd be inclined to keep the duplication on this one